src/theory/quantifiers/sygus/sygus_unif_io.cpp

```diff
@@ -1427,8 +1427,15 @@ Node SygusUnifIo::constructBestConditional(Node ce,
   // initially set minEntropy to > 1.0.
   double minEntropy = 2.0;
   unsigned bestIndex = 0;
-  for (unsigned j = 0; j < nsolved; j++)
+  for (unsigned j = 0; j < nconds; j++)
   {
+    // To compute the entropy for a condition C, for pair of terms (s, t), let
+    //   prob(t) be the probability C evaluates to t on an active point,
+    //   prob(s|t) be the probability that an active point on which C
+    //     evaluates to t has output s.
+    // Then, the entropy of C is:
+    //   sum{t}. prob(t)*( sum{s}. -prob(s|t)*log2(prob(s|t)) )
+    // where notice this is always between 0 and 1.
```

Comment:
If I understand correctly you are optimizing for conditions that lead to sets of points with the same output. This is fine if you have only T/F as output, but for the general case I think you really need to take into account the terms you are assigning to the points. This is what is done in Alur et al. TACAS'17.

Reply:
I meant that for T/F this is already optimal. But yeah, that can be addressed in a future PR. Please open an issue to keep track of it.